packages/contracts/contracts/voting/addresslist/AddresslistVoting.sol

```diff
@@ -59,61 +44,51 @@ contract AddresslistVoting is MajorityVotingBase {
     }
 
     /// @notice Checks if this or the parent contract supports an interface by its ID.
-    /// @param _interfaceId The ID of the interace.
-    /// @return bool Returns true if the interface is supported.
+    /// @param _interfaceId The ID of the interface.
+    /// @return bool Returns `true` if the interface is supported.
     function supportsInterface(bytes4 _interfaceId) public view virtual override returns (bool) {
         return
             _interfaceId == ADDRESSLIST_VOTING_INTERFACE_ID ||
             super.supportsInterface(_interfaceId);
     }
 
     /// @notice Adds new members to the address list.
-    /// @param _members The addresses of the members to be added.
+    /// @param _members The addresses of members to be added.
+    /// @dev This functin is used during the plugin initialization.
     function addAddresses(address[] calldata _members)
         external
-        auth(MODIFY_ADDRESSLIST_PERMISSION_ID)
+        auth(UPDATE_ADDRESSES_PERMISSION_ID)
     {
         _addAddresses(_members);
     }
 
-    /// @notice Internal function to add new members to the address list.
-    /// @param _members The addresses of members to be added.
-    /// @dev This functin is used during the plugin initialization.
-    function _addAddresses(address[] calldata _members) internal {
-        _updateAddresslist(_members, true);
-
-        emit AddressesAdded({members: _members});
-    }
-
-    /// @notice Removes members from the address list.
+    /// @notice Removes existing members from the address list.
     /// @param _members The addresses of the members to be removed.
     function removeAddresses(address[] calldata _members)
         external
-        auth(MODIFY_ADDRESSLIST_PERMISSION_ID)
+        auth(UPDATE_ADDRESSES_PERMISSION_ID)
     {
-        _updateAddresslist(_members, false);
-
-        emit AddressesRemoved({members: _members});
+        _removeAddresses(_members);
     }
 
-    /// @inheritdoc IMajorityVoting
+    /// @inheritdoc MajorityVotingBase
     function createProposal(
-        bytes calldata _proposalMetadata,
+        bytes calldata _metadata,
         IDAO.Action[] calldata _actions,
         uint64 _startDate,
         uint64 _endDate,
-        bool _tryEarlyExecution,
-        VoteOption _voteOption
+        VoteOption _voteOption,
+        bool _tryEarlyExecution
     ) external override returns (uint256 proposalId) {
         uint64 snapshotBlock = getBlockNumber64() - 1;
```

Comment:
This subtraction could use `unchecked`

Reply:
Done in bb6756f5cdc2fca256695bb3cdbc87161b6fd30b